docs/roadmap.md

```diff
@@ -0,0 +1,169 @@
+    `scalac.completions.enabled=true`.
+  * Requires [`metalsSetup`](#metalssetup).
+  * Shows parameter list as you type (`textDocument/signatureHelp`).
+* [ ] Auto-import global symbol from classpath.
+* [ ] Auto-expand abstract methods when implementing a trait or abstract class.
+
+## Low CPU and memory usage
+
+Heavy resource usage is one of the most frequent complaints about IDEs. Low CPU
+and memory usage becomes even more important when working in larger codebases.
+
+* [ ] Persistent symbol index: currently Metals stores the navigation index
+      in-memory, which can consume multiple GB of RAM for large projects.
+* [ ] Faster Scala and Java outline indexing. Currently, Metals can index the
+      outlines of Scala sources at ~30-40k loc/s and Java sources at ~450k
+      loc/s.
+* [ ] Throttled background indexing. Currently, all available CPUs are utilized
+      for indexing, which makes the computer slow for a period after running
+      `metalsSetup`.
+* [ ] Faster "open symbol in workspace" (`workspace/symbol`). Currently, uses a
+      naive linear search algorithm that is slow for large projects.
+
+## Solid refactoring
+
+Refactoring makes you more productive while editing code. On demand refactoring
+such as rename or formatting are great to save keystrokes for frequent
+mechanical tasks. Passive refactoring suggestions help you learn new language
+best practices.
+
+* Format (`textDocument/formatting`) with [Scalafmt][].
+  * [x] Whole file on demand (`textDocument/formatting`)
+  * [ ] Selected range (`textDocument/rangeFormatting`)
+  * [x] On save (`textDocument/willSaveWaitUntil`)
+  * [ ] As you type (`textDocument/onTypeFormatting`)
+  * Scalafmt version can be configured with `scalafmt.version="1.3.0"`.
+  * NOTE. The first time Scalafmt is invoked is slow since it requires
+    downloading the configured Scalafmt version with Coursier.
+* [x] Rename local symbol. Requires [semanticdb-scalac](#semanticdb-scalac).
+* [ ] Rename global symbol.
+* [x] Remove unused imports (`textDocument/codeActions`). Requires
+      [semanticdb-scalac](#semanticdb-scalac).
+* [ ] Move class.
+* [ ] Organize imports.
+* [ ] Insert type annotation.
+
+## Glossary
+
+### metalsSetup
+
+Task in sbt-metals to generate build metadata. Run `*:metalsSetup` to generate
+metadata for all modules in the build, including main and test sources.
+
+### semanticdb-scalac
+
+Compiler plugin that emits [SemanticDB][] files during batch compilation in the
+build tool. Overhead of the compiler plugin is ~30%, which may improve in the
+future.
+
+[semanticdb]: https://github.com/scalameta/scalameta/blob/master/semanticdb/semanticdb3/semanticdb3.md
+[index-while-building]: https://youtu.be/jGJhnIT-D2M
+[code outline]: https://marketplace.visualstudio.com/items?itemName=patrys.vscode-code-outline
+[atom-ide-ui]: https://atom.io/packages/atom-ide-ui
+[bsp]: https://github.com/scalacenter/bsp/blob/master/docs/bsp.md
```

Comment:
nitpick: `bsp` is lowercase here and uppercase above. I would stick to uppercase (similar to LSP)

Reply:
prettier formats it to lowercase for some reason, markdown formatting is surprisingly helpful IMO